tests/test_discovery.py

```diff
@@ -884,6 +879,50 @@ def test_mtls_with_provided_client_cert(
                 self.check_http_client_cert(plus, has_client_cert=use_client_cert)
                 self.assertEqual(plus._baseUrl, base_url)
 
+    @parameterized.expand(
+        [
+            ("never", "", CONFIG_DATA_WITH_WORKLOAD , REGULAR_ENDPOINT),
+            ("auto", "", CONFIG_DATA_WITH_WORKLOAD, MTLS_ENDPOINT),
+            ("always", "", CONFIG_DATA_WITH_WORKLOAD, MTLS_ENDPOINT),
+            ("never", "", CONFIG_DATA_WITHOUT_WORKLOAD, REGULAR_ENDPOINT),
+            ("auto", "", CONFIG_DATA_WITHOUT_WORKLOAD, REGULAR_ENDPOINT),
+            ("always", "", CONFIG_DATA_WITHOUT_WORKLOAD, MTLS_ENDPOINT),
+        ]
+    )
+    def test_mtls_with_provided_client_cert_unset_environment_variable(
+        self, use_mtls_env, use_client_cert, config_data, base_url
+    ):
+        """Tests that mTLS is correctly handled when a client certificate is provided.
+
+        This test case verifies that when a client certificate is explicitly provided
+        via `client_options` and GOOGLE_API_USE_CLIENT_CERTIFICATE is unset, the 
+        discovery document build process correctly configures the base URL for mTLS 
+        or regular endpoints based on the `GOOGLE_API_USE_MTLS_ENDPOINT` environment variable.
+        """
+        if hasattr(google.auth.transport.mtls, "should_use_client_cert"):
```

Comment:
Using conditionals like this is dangerous, because we could get into a state where the tests are being skipped, but look green

I suggest:
1. Check against the auth lib version, because we always expect newer versions to support should_use_client_cert (and it should alert us if they stop!)
2. use the pytest.mark.skipif decorator around the function, to automatically skip for unsupported versions

```
from google.auth import __version__ as auth_version
from google.api_core import parse_version_to_tuple

@pytest.mark.skipif(
    parse_version_to_tuple(auth_version) < (2,43,0), 
    reason="automatic mtls only enabled in google-auth<=2.43.0"
)
 def test_mtls_with_provided_client_cert_unset_environment_variable(...):
```